### PR TITLE
Add a scope to allow resetting other users' passwords

### PR DESCRIPTION
**Pull request by allisonking**
Closes https://github.com/ethyca/fides/issues/2208
Closes https://github.com/ethyca/fides/issues/2007

### Code Changes

* [x] Removes the old `user:reset-password` scope in favor of loosening up the endpoint to just always allow users to change their own passwords (as opposed to only allowing users to change their passwords in addition to needing this scope)
* [x] Adds an alembic migration to remove `user:reset-password` from every user
  * I'm not actually sure if this is the right place for it, as we can't really do a down revision for this. It's more a data migration than a db migration 
* [x] Adds a new scope `user:password-reset` (I know, the name is confusing if the old scope still exists!! I had some trouble with naming in this PR. `password-reset` / `reset-password` just seems like the right name for this action!) which allows the user with this scope to reset passwords without knowing the old one (including other users'!)
* [x] Adds a new endpoint `/user/{user_id}/force-reset-password` to allow resetting a password without knowing the old one which is scoped on `user:password-reset`
  * The [Forgot Password Cheat Sheet](https://cheatsheetseries.owasp.org/cheatsheets/Forgot_Password_Cheat_Sheet.html) says to return a consistent message for both existent and non-existent accounts. I was going to do that for this endpoint, but thought it would cause confusion if an admin reset a user's password but the user no longer existed, so it should get a 404, but instead they get a success and then they try to tell that person to log in with the new password. Overall, I think this pattern of "admin making a password to reset a user's password" doesn't really follow best practices (as we've already discussed) and is temporary, but I could also make it always return a success instead if we want. Our other user/password related endpoints also do not give a consistent message, so maybe this is just something we need to do as a chunk once we have a proper password reset flow
* [x] Tests

### Steps to Confirm

* [ ] _list any manual steps for reviewers to confirm the changes_

### Pre-Merge Checklist

* [x] All CI Pipelines Succeeded
* Documentation:
  * [ ] documentation complete, [PR opened in fidesdocs](https://github.com/ethyca/fidesdocs/pulls)
  * [ ] documentation [issue created in fidesdocs](https://github.com/ethyca/fidesdocs/issues/new/choose)
* [ ] Issue Requirements are Met
* [ ] Relevant Follow-Up Issues Created
* [x] Update `CHANGELOG.md`

### Description Of Changes

_Write some things here about the changes and any potential caveats_

### CODE REVIEW

File: src/fides/api/ctl/migrations/versions/7e218e880eaf_remove_user_reset_password_scope.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+# revision identifiers, used by Alembic.
+revision = "7e218e880eaf"
+down_revision = "d6c6c6555c86"
+branch_labels = None
+depends_on = None
+
+
+def upgrade():
+    op.execute(
+        "UPDATE fidesuserpermissions SET scopes = array_remove(scopes, 'user:reset-password')"
```
Comment (seanpreston) [inline src/fides/api/ctl/migrations/versions/7e218e880eaf_remove_user_reset_password_scope.py:20]: @allisonking and I spoke on a call about when `array_remove(...)` was introduced to Postgres — it was introduced in version 11. Fides has only ever shipped with Postgres 12 and beyond so using this function shouldn't be an issue.

File: src/fides/api/ops/api/v1/endpoints/user_endpoints.py [hunk 2]
```diff
@@ -113,6 +118,32 @@ def update_user_password(
...
+    status_code=HTTP_200_OK,
+)
+def update_other_user_password(
+    *,
+    db: Session = Depends(deps.get_db),
+    user_id: str,
+    data: UserForcePasswordReset,
+) -> FidesUser:
+    """
+    Update a user's password given a `user_id`.
```
Comment (seanpreston) [inline src/fides/api/ops/api/v1/endpoints/user_endpoints.py:133]: Please can you add more context to the docstring here — it's important to mention why this exists separately to the endpoint above.

**Review by seanpreston — APPROVED** [review]
Thanks @allisonking — just one minor comment on the docstring :)

**Comment by allisonking** [comment]
Going to merge, as the failing pylint test is unrelated to this PR

**Pull request merged**